fn/option.go

```diff
@@ -26,6 +26,17 @@ func None[A any]() Option[A] {
 	return Option[A]{}
 }
 
+// MaybeSome constructs an option from a pointer.
+//
+// MaybeSome : *A -> Option[A].
+func MaybeSome[A any](a *A) Option[A] {
```

Comment:
This function is good. However, please rename it to something approximating the below list. "Maybe" does not have precedent to mean anything and I think there are more intuitive names for this signature.

- `OptionFromPtr`
- `PtrToOption`
- `DerefOrNone`

Reply:
I've gone with `OptionFromPtr` in the latest commits.

Thanks for taking a look.